Cleanup Annoy index (#50312)
ClickHouse/ClickHouse: docs/en/engines/table-engines/mergetree-family/annindexes.md

``` sql
SELECT *
FROM table
WHERE L2Distance(column, Point) < MaxDistance
LIMIT N
```

``` sql
SELECT *
FROM table
ORDER BY L2Distance(column, Point)
LIMIT N
```

The queries are expensive because the L2 (Euclidean) distance between `Point` and all points in `column` and must be computed. To speed this process up, Approximate Nearest Neighbor Search Indexes (ANN indexes) store a compact representation of the search space (using clustering, search trees, etc.) which allows to compute an approximate answer quickly.

# Creating ANN Indexes

As long as ANN indexes are experimental, you first need to `SET allow_experimental_annoy_index = 1`.

Syntax to create an ANN index over an `Array` column:

```sql
CREATE TABLE table
(
  `id` Int64,
  `embedding` Array(Float32),
  INDEX <ann_index_name> embedding TYPE <ann_index_type>(<ann_index_parameters>) GRANULARITY <N>
)
ENGINE = MergeTree
ORDER BY id;
```

Syntax to create an ANN index over a `Tuple` column:

```sql
CREATE TABLE table
(
  `id` Int64,
  `embedding` Tuple(Float32[, Float32[, ...]]),
  INDEX <ann_index_name> embedding TYPE <ann_index_type>(<ann_index_parameters>) GRANULARITY <N>
)
ENGINE = MergeTree
ORDER BY id;
```

> ANN indexes are built during column insertion and merge and `INSERT` and `OPTIMIZE` statements will slower than for ordinary tables. ANNIndexes are ideally used only with immutable or rarely changed data, respectively comparatively many more read requests than write requests.

Reviewer: ```suggestion
ANN indexes are built during column insertion and merge and `INSERT` and `OPTIMIZE` statements will be slower than for ordinary tables. ANNIndexes are ideally used only with immutable or rarely changed data, respectively there are much more read requests than write requests.
```
